### PR TITLE
UT2004 Revival

### PR DESCRIPTION
**Pull request by SoapGentoo**

### CODE REVIEW

File: games-fps/ut2004-bonuspack-mega/ut2004-bonuspack-mega-1-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+KEYWORDS="~amd64 ~x86"
+
+BDEPEND="app-arch/unzip"
+
+S="${WORKDIR}"
+
+src_prepare() {
+	default
+
+	mv -f UT2004MegaPack/* . || die
```
Comment (mgorny) [inline games-fps/ut2004-bonuspack-mega/ut2004-bonuspack-mega-1-r3.ebuild]: Why not change S instead?

File: games-fps/ut2004-bonuspack-ece/ut2004-bonuspack-ece-1-r4.ebuild [hunk 2]
```diff
@@ -54,3 +59,10 @@ src_prepare() {
 	rm System/{Core.int,Engine.int,Setup.int,Window.int} || die
 	rm System/{OnslaughtFull.u,OnslaughtBP.u} || die
 }
+
+src_install() {
+	insinto /opt/ut2004
+	doins -r Animations Help Maps Sounds StaticMeshes System Textures
```
Comment (mgorny) [inline games-fps/ut2004-bonuspack-ece/ut2004-bonuspack-ece-1-r4.ebuild]: Wouldn't `*` work here? Especially that there's a lot of `rm`-ing anyway.

File: games-fps/ut2004-bonuspack-mega/ut2004-bonuspack-mega-1-r3.ebuild [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	rm Help/{ReadMePatch.int.txt,UT2004Logo.bmp} || die
+	mv Help/BonusPackReadme.txt Help/MegapackReadme.txt || die
+
+	rm Maps/ONS-{Adara,IslandHop,Tricky,Urban}.ut2 || die
+	rm Sounds/{CicadaSnds,DistantBooms,ONSBPSounds}.uax || die
+	rm StaticMeshes/{BenMesh02,BenTropicalSM01,HourAdara,ONS-BPJW1,PC_UrbanStatic}.usx || die
+
+	# System
+	rm System/{AL,AS-,B,b,C,D,E,F,G,I,L,O,o,S,s,U,V,W,X,x}* || die
```
Comment (mgorny) [inline games-fps/ut2004-bonuspack-mega/ut2004-bonuspack-mega-1-r3.ebuild]: Yyyyyy, wouldn't a whitelist (i.e. installing files selectively) work better here?

File: games-fps/ut2004-data/ut2004-data-3186-r5.ebuild [hunk 4]
```diff
@@ -64,17 +63,35 @@ grabdirs() {
...
+	if [[ -f ${ut_unpack} ]] ; then
+		uz2unpack "${ut_unpack}" "${ut_unpack%.uz2}" \
+			|| die "uncompressing file ${ut_unpack}"
+	fi
+	if [[ -d ${ut_unpack} ]] ; then
+		while read f ; do
+			uz2unpack "${ut_unpack}/${f}" "${ut_unpack}/${f%.uz2}" \
+				|| die "uncompressing file ${f}"
+			rm -f "${ut_unpack}/${f}" || die "deleting compressed file ${f}"
+		done < <(find "${ut_unpack}" -maxdepth 1 -name '*.uz2' -printf '%f\n' 2>/dev/null)
```
Comment (mgorny) [inline games-fps/ut2004-data/ut2004-data-3186-r5.ebuild]: Why not `-print0`?

File: games-fps/ut2004-data/ut2004-data-3186-r5.ebuild [hunk 5]
```diff
@@ -294,84 +312,86 @@ src_install() {
...
 	# Create empty files in Benchmark
 	for j in {CSVs,Logs,Results} ;do
-		keepdir "${dir}"/Benchmark/${j}
+		keepdir /opt/ut2004/Benchmark/${j}
 	done
 
-	make_wrapper ut2004 ./ut2004-bin "${dir}"/System "${dir}"/System "${dir}"
+	make_wrapper ut2004 ./ut2004-bin /opt/ut2004/System /opt/ut2004/System /opt/ut2004
 
 	# Remove unneccessary files
```
Comment (mgorny) [inline games-fps/ut2004-data/ut2004-data-3186-r5.ebuild]: You realize that keeping a huge blacklist is just insane?

**Comment by chewi** [comment]
@mgorny, I think all the things you've commented on are issues with the existing ebuild. I think it could be a lot simpler but it's a big job we don't have time for before the last-rite. I may look into it afterwards as I have interest in this game and I do have the DVD. As things stand, it does install for me.

**Comment by chewi** [comment]
@SoapGentoo, what are you doing about the ut2004-ded package? The `/opt/ut2004-ded` symlink you have created in the bonuspack packages will collide with this. Furthermore, I don't want to have to create that symlink in all the other mod packages. Could we just have `ut2004-ded` and `ut2004` block each other entirely (instead of just `ut2004[dedicated]`) so that we can install them to the same place and avoid this whole mess?

File: games-fps/ut2004/ut2004-3369.3-r2.ebuild [hunk 6]
```diff
@@ -1,14 +1,16 @@
...
+inherit desktop eutils
+
+MY_P="${PN}-lnxpatch${PV%.*}-2.tar.bz2"
 
-MY_P="ut2004-lnxpatch${PV%.*}-2.tar.bz2"
 DESCRIPTION="Editor's Choice Edition plus Mega Pack for the well-known first-person shooter"
 HOMEPAGE="http://www.unrealtournament2004.com/"
 SRC_URI="
-	http://treefort.icculus.org/${PN}/${MY_P}
+	http://ut2004.ut-files.com/index.php?dir=Patches/Linux/&file=${MY_P} -> ${MY_P}
```
Comment (chewi) [inline games-fps/ut2004/ut2004-3369.3-r2.ebuild]: This can just be `https://ut2004.ut-files.com/Patches/Linux/${MY_P}`.

**Comment by chewi** [comment]
This is now merged with a bunch of tweaks. Thanks!

**Pull request closed without merging**